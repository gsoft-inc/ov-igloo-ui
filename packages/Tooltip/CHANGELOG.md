# @igloo-ui/tooltip

## 3.1.2

### Patch Changes

- bf478b5: Added missing tests. Fixed missing code in stories when clicking 'Show code'. Updated Dropdown mock. Added new optional prop highlightToday to help with test snapshots.

## 3.1.1

### Patch Changes

- 49d3c6b: Added a mock for testing

## 3.1.0

### Minor Changes

- bf0342e: Refactor: Replaced react spring and react popper with framer motion and/or floating-ui

### Patch Changes

- 9ed10b2: Updated jest and updated snapshots for popover and tooltip components

## 3.0.3

### Patch Changes

- 4230499: Fix: updated components to use the IconButton with appearance ghost, variant secondary so that they are grey like they used to be.

## 3.0.2

### Patch Changes

- a3e8f88: Updated the variables used for the light theme of the tooltip

## 3.0.1

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 3.0.0

### Major Changes

- ff6db5c: change of behavior for the tooltip when scrolling and displaying in the mobile version.

  ## BREAKING CHANGE

  - disabling the tooltip in mobile devices. It is however possible to display it by passing the props `showOnMobile`

## 2.0.2

### Patch Changes

- 0640409: udpate classnames to 2.3.2

## 2.0.1

### Patch Changes

- 7cf9a4c: Adjusted container display to make sure it works for block elements

## 2.0.0

### Major Changes

- 17def56: Added animation to tooltip

  ## BREAKING CHANGE

  Changed the default positioning to 'top' instead of 'auto'

## 1.0.5

### Patch Changes

- 088f9fd: Fix: Changed the mouseleave event to use native js so it works with disabled elements as children.

## 1.0.4

### Patch Changes

- 4092ee6: center text when the position is `top` or `bottom`

## 1.0.3

### Patch Changes

- d15162f: Change position of Tooltip from absolut to fixed

## 1.0.2

### Patch Changes

- 9eab44f: Fix the maxWidth

## 1.0.1

### Patch Changes

- 0a86f13: Update content of README

## 1.0.0

### Major Changes

- 9fc50a4: Use popper for the postion of the tooltip

  ## BREAKING CHANGE

  removing the `forwardref`

## 0.0.9

### Patch Changes

- 01fbda0: add posibility to forward a `ref`

## 0.0.8

### Patch Changes

- 8ceb764: fix error with calc() on the transition of arrow

## 0.0.7

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.0.6

### Patch Changes

- 81f7d46: show tooltip on overflow content

## 0.0.5

### Patch Changes

- 4a256e3: IE11 compatibility

## 0.0.4

### Patch Changes

- 56816b9: Replace the background color by `--electric-blue-900`

## 0.0.3

### Patch Changes

- 67f63b7: Fix Tooltip position for children other then text

## 0.0.2

### Patch Changes

- 2da8010: Fix render of elements with the css var --grey-{...}
