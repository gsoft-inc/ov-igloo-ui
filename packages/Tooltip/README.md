# @igloo-ui/tooltip

Welcome to my `Tooltip` component.

## Installation

```sh
npm i @igloo-ui/tooltip
# or with yarn
yarn add @igloo-ui/tooltip
```

Then to use the component in your code just import it!

```js
import COMPONENT_NAME from '@igloo-ui/tooltip';
// and with css-modules
import '@igloo-ui/tooltip/dist/tooltip.css';
```
