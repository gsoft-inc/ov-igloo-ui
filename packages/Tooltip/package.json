{
  "name": "@igloo-ui/tooltip",
  "version": "1.0.5",
  "main": "dist/Tooltip.js",
  "module": "dist/Tooltip.js",
  "types": "dist/Tooltip.d.ts",
  "styles": "dist/tooltip.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com",
    "Alexande alexandre.beaudoin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6",
    "react-dom": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^1.0.0",
    "@popperjs/core": "^2.11.5",
    "classnames": "^2.3.1",
    "react-popper": "^2.3.0",
    "react-spring": "9.5.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
