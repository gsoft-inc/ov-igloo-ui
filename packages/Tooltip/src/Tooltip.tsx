import * as React from 'react';
import classNames from 'classnames';

import './tooltip.scss';

export type Appearance = 'dark' | 'light';
export type Position = 'top' | 'right' | 'bottom' | 'left';

export interface TooltipProps extends React.ComponentProps<'div'> {
  children: string | React.ReactNode;
  tooltipClassName?: string;
  // content?: React.ReactNode;
  position?: Position;
  appearance?: Appearance;
  maxWidth?: number;
  spacing?: number;
  arrowVisible?: boolean;
}

const Tooltip: React.FunctionComponent<TooltipProps> = (
  props: TooltipProps
) => {
  const {
    children,
    tooltipClassName,
    position = 'top',
    appearance = 'dark',
    maxWidth = 200,
    spacing = 8,
    arrowVisible = true,
    className,
    ...rest
  } = props;

  const classes = classNames('ids-tooltip', className, {
    [`ids-tooltip--${position}`]: true,
    'has-arrow': arrowVisible,
    'ids-tooltip-dark': appearance === 'dark',
  });

  return (
    <div className={classes} {...rest}>
      {children}
    </div>
  );
};

export default Tooltip;
