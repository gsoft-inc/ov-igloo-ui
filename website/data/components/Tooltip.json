[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Tooltip/src/Tooltip.tsx",
      "description": "",
      "displayName": "Tooltip",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "The target button, text, svg etc.. of the Tooltip.",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "tooltipClassName": {
          "defaultValue": null,
          "description": "Add a specific class to the tooltip",
          "name": "tooltipClassName",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "content": {
          "defaultValue": null,
          "description": "The content to display inside the tooltip",
          "name": "content",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "position": {
          "defaultValue": null,
          "description": "The position the tooltip is on.",
          "name": "position",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "Position" }
        },
        "appearance": {
          "defaultValue": null,
          "description": "The appearance of the tooltip.",
          "name": "appearance",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "Appearance" }
        },
        "maxWidth": {
          "defaultValue": null,
          "description": "The max with of the tooltip",
          "name": "maxWidth",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "number" }
        },
        "active": {
          "defaultValue": null,
          "description": "When True, manually show the Tooltip.",
          "name": "active",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "disabled": {
          "defaultValue": null,
          "description": "When True, disabled the tooltip",
          "name": "disabled",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
            "name": "TooltipProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Tooltip/src/Tooltip.tsx",
              "name": "TooltipProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
