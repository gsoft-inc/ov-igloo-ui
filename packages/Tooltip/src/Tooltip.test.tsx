/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import Tooltip, { TooltipProps } from './Tooltip';

const setUp = (props: TooltipProps) => {
  const component = shallow(<Tooltip {...props}>tooltip</Tooltip>);
  return component;
};

describe('Tooltip', () => {
  let component: any;
  beforeEach(() => {
    component = setUp({ children: 'children' });
  });

  test('It should render without errors', (): void => {
    const container = component.find('.ids-tooltip');
    expect(container.length).toBe(1);
  });

  test('It should render a snapshot', (): void => {
    expect(component).toMatchSnapshot();
  });

  test('It should render at Top', (): void => {
    const tooltip = setUp({ position: 'top', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip--top');

    expect(bubble.length).toBe(1);
  });

  test('It should render at Right', (): void => {
    const tooltip = setUp({ position: 'right', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip--right');

    expect(bubble.length).toBe(1);
  });

  test('It should render at Bottom', (): void => {
    const tooltip = setUp({ position: 'bottom', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip--bottom');

    expect(bubble.length).toBe(1);
  });

  test('It should render at Left', (): void => {
    const tooltip = setUp({ position: 'left', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip--left');

    expect(bubble.length).toBe(1);
  });

  test('It should render using light theme', (): void => {
    const tooltip = setUp({ appearance: 'light', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip-dark');

    expect(bubble.length).toBe(0);
  });

  test('It should render using dark theme', (): void => {
    const tooltip = setUp({ appearance: 'dark', children: 'children' });
    const bubble = tooltip.find('.ids-tooltip-dark');

    expect(bubble.length).toBe(1);
  });
});
