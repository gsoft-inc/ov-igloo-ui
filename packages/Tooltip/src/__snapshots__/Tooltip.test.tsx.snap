// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip It should render a snapshot 1`] = `ShallowWrapper {}`;
